### PR TITLE
[Search-183] Add tests

### PR DESCRIPTION
**Pull request by MBkkt**
### Testing & Verification

- [x] This PR adds tests that were used to verify all changes:
  - [x] Added new C++ **Unit tests**

### CODE REVIEW

File: tests/IResearch/IResearchLinkTest.cpp [hunk 1]
```diff
@@ -2581,7 +2589,7 @@ TEST_F(IResearchLinkMetricsTest, LinkAndMetics) {
     std::string actual;
     getPrometheusStr(actual);
 
-    EXPECT_EQ(actual, expected);
+    EXPECT_TRUE(actual == expected);
```
Comment (Dronplane) [inline tests/IResearch/IResearchLinkTest.cpp]: Why not leave EXPEC_EQ  here ?

File: tests/IResearch/IResearchLinkTest.cpp [hunk 2]
```diff
@@ -2537,7 +2545,7 @@ TEST_F(IResearchLinkMetricsTest, LinkAndMetics) {
...
```
Comment (Dronplane) [inline tests/IResearch/IResearchLinkTest.cpp]: Why not leave EXPEC_EQ  here ?

**Review by Dronplane — CHANGES_REQUESTED** [review]
LGTM in general, but couple of places looks suspicious

File: tests/IResearch/IResearchLinkTest.cpp (hunk 1 above)
Comment (MBkkt) [reply tests/IResearch/IResearchLinkTest.cpp]: Just typo

File: tests/IResearch/IResearchLinkTest.cpp (hunk 2 above)
Comment (MBkkt) [reply tests/IResearch/IResearchLinkTest.cpp]: same here

**Review by Dronplane — APPROVED** [review]
LGTM

**Pull request closed without merging**